bake/bake.go

```diff
@@ -245,13 +246,28 @@ func ParseFiles(files []File, defaults map[string]string) (_ *Config, err error)
 	}
 
 	if len(hclFiles) > 0 {
-		if err := hclparser.Parse(hcl.MergeFiles(hclFiles), hclparser.Opt{
+		renamed, err := hclparser.Parse(hcl.MergeFiles(hclFiles), hclparser.Opt{
 			LookupVar:     os.LookupEnv,
 			Vars:          defaults,
 			ValidateLabel: validateTargetName,
-		}, &c); err.HasErrors() {
+		}, &c)
+		if err.HasErrors() {
 			return nil, err
 		}
+
+		for _, renamed := range renamed {
```

Comment:
Does this need to be outside of the `hclparser` pkg? Why not just do it as part of `Parse()` ?

Reply:
Yeah, this is unfortunately required. The `bake/hclparser` package parses into a generic struct type, so doesn't actually have any knowledge of `Target`s or `Group`s, which theoretically would make it reusable in other places if we ever wanted to (or would make adding a new block type simple).

Only the `bake` package actually has knowledge of the `Target` and `Group`, and how they interact, so we need the logic here. :cry: